compass/landice/tests/mismipplus/setup_mesh.py

```diff
@@ -0,0 +1,285 @@
+        ds_mesh[f'x{loc}'] = ds_mesh[f'x{loc}'] + xShift
+        ds_mesh[f'y{loc}'] = ds_mesh[f'y{loc}'] + yShift
+
+    # WHL :
+    #   Need to adjust geometry along top and bottom boundaries to get flux
+    #   correct there. Essentially, we only want to model the interior half of
+    #   those cells. Adding this here because we only want to do this if it
+    #   hasn't been done before. This method is assuming a periodic_hex mesh!
+
+    # Reduce the cell areas by half along the N/S boundary
+
+    # Boolean maks for indexes which correspond to the N/S boudnary of mesh
+    mask = ((ds_mesh.yCell == ds_mesh.yCell.min()) |
+            (ds_mesh.yCell == ds_mesh.yCell.max()))
+
+    ds_mesh['areaCell'] = xr.where(mask,
+                                   ds_mesh.areaCell * 0.5,
+                                   ds_mesh.areaCell)
+
+    # Reduce the edge lengths by half along the N/S boundary
+    unique_ys_edge = np.unique(ds_mesh.yEdge.data)
+
+    # Boolean maks for indexes edges on the N/S boudnary of mesh
+    mask = ((ds_mesh.yEdge == unique_ys_edge[0]) |
+            (ds_mesh.yEdge == unique_ys_edge[-1]))
+    # WHL: zero out the edges on the boundary
+    #      (not necessary because velocity will also be zero)
+    ds_mesh['dvEdge'] = xr.where(mask, 0.0, ds_mesh.dvEdge)
+
+    # Boolean mask for the indexed of edges between N/S boundary cells
+    mask = ((ds_mesh.yEdge == unique_ys_edge[1]) |
+            (ds_mesh.yEdge == unique_ys_edge[-2]))
+    # cut length in half for edges between boundary cells
+    ds_mesh['dvEdge'] = xr.where(mask, ds_mesh.dvEdge * 0.5, ds_mesh.dvEdge)
+
+    return ds_mesh
+
+
+def __mismipplus_bed(x, y):
+    x = x / 1.e3      # m to km
+    y = y / 1.e3      # m to km
+    B0 = -150.        # m
+    B2 = -728.8       # m
+    B4 = 343.91       # m
+    B6 = -50.57       # m
+    x_bar = 300.      # km
+    x_tilde = x / x_bar
+    dc = 500.         # m
+    fc = 4.           # km
+    wc = 24.          # km
+    Ly = 80.          # km
+    Bmax = -720.      # m
+
+    B_x = B0 + B2 * x_tilde**2 + B4 * x_tilde**4 + B6 * x_tilde**6
+    B_y = dc / (1 + np.exp(-2 * (y - Ly / 2 - wc) / fc)) +\
+        dc / (1 + np.exp(2 * (y - Ly / 2 + wc) / fc))
+
+    Bsum = B_x + B_y
+
+    z_b = np.maximum(Bsum, Bmax)  # Eqn 1. (Asay-Davis et al. 2016)
+
+    return z_b
+
+
+def _setup_MISMPPlus_IC(config, logger, filename):
+    """
+    Add the inital condition for the MISMIP+ spinup to the given MPAS mesh file
+
+    Parameters
+    ----------
+    config : comass.config.CompassConfigParser
+        Configuration options for this test case, ....
+
+    logger : logging.Logger
+        A logger for output from the step
+
+    filename : str
+        NetCDF file to place the MISMIP+ ICs in
+
+    """
+
+    # Hard code some parameters from Table 1. of Asay-Davis et al. 2016
+    accum = 0.3       # m^{-1}
+    C = 3.160e6       # Pa m^{-1/3} s^{1/3}
+    rhoi = 918        # kg m^{-3}
```

Comment:
I'm happy to use 910, but Table 1 from [Asay-Davis et al. 2016](https://gmd.copernicus.org/articles/9/2471/2016/gmd-9-2471-2016.pdf) has 918. So I guess if we were strictly following MISMIP+ we'd use that. I guess the difference would be _really_ small in the actual simulation results. 

Since this test case is for other modelling purposes besides the actual MISMIP+ experiments, should I move it to the config file and set at 910, but add a comment that we are departing from MISMIP+ conventions? 

That way if at some point there was desire to actually run MISMIP+ there wouldn't be need to change any code?

Reply:
I like the idea of making rhoi a cfg option for this particular test case.  It is important to be able to reproduce the prescribed MISMIP+ protocol if needed, but we otherwise prefer not to use pure ice density.  Putting it in a cfg where it can be used to automatically update both the initial condition calculations and the namelist is the safest, most flexible approach, I think.